### ISSUE TITLE
tests: sstables_3_x_test fails due to compression library differences (s390x)

### ISSUE THREAD

**Issue by mundaym** [scylladb/scylla#3923]
Some tests in sstable_3_x are failing in master (9e4276669b5ac4d66f0bea6c993d702a0156fc81) and in 2.3.1. It looks like most of the failing tests use compression. I'm not sure if the failures are indicative of a real issue with compression or just a mismatch with files generated on a different system with a different version of the relevant compression libraries installed. Some of the output seems to include some of the same bytes but in a different order so this might be an endianess issue somewhere. For example:

```
tests/sstable_3_x_test.cc(3019): ESC[1;31;49merror: in "test_write_many_partitions_lz4": check { b1, e1 } == { b2, e2 } has failed. 
Mismatch at position 664: 0x1e != '-'
Mismatch at position 0x29b: 0x2 != 0x13
Mismatch at position 0x29c: 0x1e != 0
Mismatch at position 0x29d: 0x97 != 0x1e
Mismatch at position 0x29e: 0xbe != 0x97
Mismatch at position 0x29f: 0 != 0xbe
Mismatch at position 0x2a0: '-' != 0
Mismatch at position 0x2a1: 0xe6 != '-'
Mismatch at position 0x2a2: 0xa3 != 0xe6
Mismatch at position 0x2a3: '9' != 0xa3
Mismatch at position 0x2a4: 0 != '&'
Mismatch at position 0x2a5: '-' != 0
Mismatch at position 0x2a6: 0xf5 != '-'
Mismatch at position 0x2a7: '`' != 0xf5
Mismatch at position 0x2a8: 0x13 != '`'
Mismatch at position 0x2a9: 0 != 0x13
Mismatch at position 0x2aa: '-' != 0
Mismatch at position 0x2ab: 'k' != '-'
Mismatch at position 0x2ac: 0x1f != 'k'
Mismatch at position 0x2ad: 0x13 != 0x1f
Mismatch at position 0x2ae: 0 != 0x13
Mismatch at position 0x2af: '-' != 0
<continued>
```

Here is a cut down version of the test output:

```
tests/sstable_3_x_test.cc(3019): error: in "test_write_many_partitions_lz4": check { b1, e1 } == { b2, e2 } has failed. 
tests/sstable_3_x_test.cc(3019): error: in "test_write_many_partitions_lz4": check { b1, e1 } == { b2, e2 } has failed. 
tests/sstable_3_x_test.cc(3019): error: in "test_write_many_partitions_lz4": check { b1, e1 } == { b2, e2 } has failed. 
tests/sstable_3_x_test.cc(3019): error: in "test_write_many_partitions_snappy": check { b1, e1 } == { b2, e2 } has failed. 
tests/sstable_3_x_test.cc(3019): error: in "test_write_many_partitions_snappy": check { b1, e1 } == { b2, e2 } has failed. 
tests/sstable_3_x_test.cc(3019): error: in "test_write_many_partitions_snappy": check { b1, e1 } == { b2, e2 } has failed. 
unknown location(0): fatal error: in "test_write_large_clustering_key": sstables::malformed_sstable_exception: end of input, but not end of partition

7 failures are detected in the test module "tests/sstable_3_x_test.cc"
```

**Comment by argenet** [comment]
Thanks for reporting the issue.

I have reproduced it on Ubuntu 18.04 with the failing `test_write_many_partitions_lz4` and found that the files produced by the test, although binary-different from the originals, are valid and dumped fine with `sstabledump`. This likely hints at some compression-level difference that does not affect the data validity.

I will do more checks to ensure that there is no danger of data corruption, likely we'll need to remove the byte-by-byte check from the failing tests on compressed SSTables 3.x files and only leave write-then-read validation.

**Comment by avikivity** [comment]
Certainly we can't compare compressed files byte-by-byte. But we can decompress and compare.

**Comment by argenet** [comment]
This has been addressed in `master` with commit a29c9189cbb041bfc1b7647837999875886da031.
